### PR TITLE
[YouTube] Support Shorts UI in playlists

### LINKED ISSUE #1077: [YouTube] Cannot get streams of playlists with only shorts
With the rollout of the sort by oldest feature on the new filtering system, YouTube also added filters to filter playlists, allowing users to filter videos and shorts inside playlists.

When shorts filter is enabled or when there are only shorts in a playlist, YouTube return shorts UI elements instead of standard video ones, the ones that are also used for shorts shelves in searches and suggestions and shorts in the corresponding channel tab.

Short UI elements (`reelItemRenderer`s) don't include any information about shorts' upload date, shorts' uploader name and shorts' uploader URL; their duration needs to be parsed from accessibility data and their view count extracted is not exact.

The data structure has changed, switching from `playlistVideoListRenderer`s and `playlistVideoRenderer`s to `richGridRenderer`s and `richItemRenderer`s + `reelItemRenderer`s respectively), so extraction of playlists in this case returns no streams. `reelItemRenderer`s are also not supported by the extractor.

The shorts playlist UI doesn't give any continuation when there are more than 100 shorts in a playlist. This seems to be a bug on YouTube's side.

I think we have three possibilities to fix or mitigate the issues these changes raised:
- ~~use a mobile client (`MWEB`, `ANDROID` and probably `IOS`), on which filters are available but video UI is returned by default when there are only shorts in a playlist;~~ (edit: shorts UI returned and forced for some playlists)
- use the `TVHTML5` client, where no filters and no precise view count are available (only human ones such as `1k views` are returned);
- keep the desktop variant and support `reelItemRenderer`s, even if we loose a lot of information.

### PR DESCRIPTION
**Pull request by AudricV**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [ ] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
  I only used my test class on which no errors occurred during my tests.

This PR fixes the extraction of items in YouTube playlists returning a Shorts UI.

A new test in `YoutubePlaylistExtractorTest` has been added to test the extraction of items with this playlist UI, which uses the system playlist of uploaded shorts by the official YouTube channel (`UUSHBR8-60-B28hp2BmDPdntcQ`).

As one of the effects of YouTube using a shorts UI, no uploader info is available for each stream. The default tests expected that the uploader name was set, so I had to remove this check (I also removed the uploader URL one, as it was commented out).

Finally this PR removes an outdated A/B test comment in `YoutubePlaylistExtractor` and fixes a bug in the `assertNotEmpty` test method of the extractor's assertions class (`ExtractorAsserts`) which tested that the test failure message, annotated as nullable, was not null instead of the string to check (it was the initial cause of the stream items' tests failures of my test class added with this PR, as no uploader name was provided and a `NullPointerException` was thrown).

Fixes #1077.

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java [hunk 1]
```diff
@@ -327,17 +326,22 @@ public InfoItemsPage<StreamInfoItem> getInitialPage() throws IOException, Extrac
                 .map(content -> content.getObject("itemSectionRenderer")
                         .getArray("contents")
                         .getObject(0))
-                .filter(contentItemSectionRendererContents ->
-                        contentItemSectionRendererContents.has(PLAYLIST_VIDEO_LIST_RENDERER)
-                                || contentItemSectionRendererContents.has(
-                                "playlistSegmentRenderer"))
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java:333]: Nice catch that "playlistSegmentRenderer" wasn't really used even before!

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java [hunk 2]
```diff
@@ -399,14 +403,26 @@ private Page getNextPageFrom(final JsonArray contents)
...
+                        if (richItemRenderer.has("content")) {
+                            final JsonObject richItemRendererContent =
+                                    richItemRenderer.getObject("content");
+                            if (richItemRendererContent.has(REEL_ITEM_RENDERER)) {
+                                collector.commit(new YoutubeReelInfoItemExtractor(
+                                        richItemRendererContent.getObject(REEL_ITEM_RENDERER),
+                                        timeAgoParser));
+                            }
+                        }
+                    }
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java:424]: Shouldn't an error be added to the collector to signal that an item in the list was in an unrecognized format? Or is there other garbage inside the list that just needs to be ignored?

**Review by Stypox — CHANGES_REQUESTED** [review]
LGTM, thanks! Just one thing to check

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java (hunk 1 above)
Comment (AudricV) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java:333]: It seems it was used for seasonal playlists: see https://github.com/TeamNewPipe/NewPipeExtractor/commit/8b3f90eb7e4a56ee7e33128af7cf8dd94663414a.

I don't remember now why I have added this though.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java (hunk 2 above)
Comment (AudricV) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java:424]: I am not sure, as there should be playlists video items only but YouTube return(ed?) suggestions of videos to add to users playlists when their owner(s) browse the playlists and is(are) logged with their account(s), so I don't know what elements they use in this case.

Comment (Stypox) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java:424]: Ok let's leave it this way and if users report of missing item take a look again

**Review by Stypox — APPROVED** [review]

**Pull request merged**